### PR TITLE
add constructor for codegen to set flatpackage

### PR DESCRIPTION
**Pull request by borissmidt**

### CODE REVIEW

File: grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package monix.grpc
+
+import protocbridge.{Artifact, SandboxedJvmGenerator}
+import scalapb.GeneratorOption
+import scalapb.GeneratorOption.FlatPackage
+
+/**
```
Comment (jvican) [inline grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala:7]: Remove?

File: grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  * Copyright (C) 08.11.21 - REstore NV
+  */
+
+package object codegen {
+  val SandboxedGenerator = SandboxedJvmGenerator.forModule(
```
Comment (jvican) [inline grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala:12]: Make it private?

File: grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import protocbridge.{Artifact, SandboxedJvmGenerator}
+import scalapb.GeneratorOption
+import scalapb.GeneratorOption.FlatPackage
+
+/**
+  * Copyright (C) 08.11.21 - REstore NV
+  */
+
+package object codegen {
```
Comment (jvican) [inline grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala:11]: Why a package object?

File: grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    Artifact(
+      "me.vican.jorge",
+      "monix-grpc-codegen_2.12",
+      BuildInfo.version
+    ),
+    "scalapb.ScalaPbCodeGenerator$",
+    GrpcCodeGenerator.suggestedDependencies
+  )
+
+  def apply(options: Set[GeneratorOption]): (SandboxedJvmGenerator, Seq[String]) =
```
Comment (jvican) [inline grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala:23]: Looks like we're going to support other options in the future, I suggest we use the builder pattern instead and remove the need for all these redundant `apply` methods with default arguments. Will simplify the API and will remove the need to return a tuple.

**Review by jvican — CHANGES_REQUESTED** [review]

File: grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala (hunk 3 above)
Comment (borissmidt) [reply grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala:11]: It is a copy of the codegen package object in scalapb. I could make it a lower cased name object.

**Comment by borissmidt** [comment]
Looking at the commments of yours, it is a copy of the codegen object of the scalapb generator and it is meant to be called from sbt directly. 
So i made is as consistent as possible.

The tuple is just the way the scalapb api works, i.e. jar + arguments

File: grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala (hunk 4 above)
Comment (borissmidt) [reply grpc-codegen/src/main/scala/monix/grpc/codegen/package.scala:20]: The tupple is what we have to return to the scalapb plugin part of sbt. So we can't go around that. (i.e. it is passed like strings)

**Pull request closed without merging**